include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,338 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
```

Comment:
I think we need to improve the commit message a bit. 

What is the difference between a component and a module? There is no information on pipeline operations, so whymention it? 

Why talk about pipeline operations when there's no mention on them in the code? and why do we need to convert IPC4 to IPC3? Jst not following this statement.

Finally, what does "Module interface can also be supported by current component interface" really mean? Am I the only one struggling to understand this?